pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1616,6 +1647,50 @@ func configAdapters(ctx *domainContext, config types.DomainConfig) error {
 	return nil
 }
 
+func writeMountPointsToFile(mountPoints []MountPoints, status types.DomainStatus, file *os.File) error {
+	if len(mountPoints) != len(status.DiskStatusList){
+		err := fmt.Errorf("writeMountPointsToFile: Number of source: %v and target: %v mismatch.",
+			len(status.DiskStatusList), len(mountPoints))
+		log.Errorf(err.Error() )
+		return err
+	}
+
+	for i, mp := range mountPoints {
+		if mp.fileSystem == "" {
+			mp.fileSystem = "vfat"
+		} else {
+			if _, ok := FileSystems[mp.fileSystem]; !ok {
+				err := fmt.Errorf("writeMountPointsToFile: Invalid/unrecognized filesystem: %v", mp.fileSystem)
+				log.Errorf(err.Error())
+				return err
+			}
+		}
+		if mp.partition == 0 { //Assuming if mp.partition = 0, then it is not set
+			mp.partition = 1
+		} else {
+			if mp.partition > 4 || mp.partition < 0 {
+				err := fmt.Errorf("writeMountPointsToFile: Invalid partition: %v. Must be between 1 - 4", mp.partition)
+				log.Errorf(err.Error())
+				return err
+			}
+		}
+		if mp.targetPath == "" {
+			err := fmt.Errorf("writeMountPointsToFile: targetPath cannot be empty")
+			log.Errorf(err.Error())
+			return err
+		} else if !strings.HasPrefix(mp.targetPath, "/") {
+			//Target path is expected to be absolute.
+			err := fmt.Errorf("writeMountPointsToFile: targetPath should be absolute")
+			log.Errorf(err.Error())
+			return err
+		}
```

Comment:
How do u get these target paths?

Reply:
Restructured it to get it from AppManifest.